action.yml

```diff
@@ -0,0 +1,137 @@
+name: "GET Toolkit to lint and notify failures"
+description: "Lints using Dockle and ArgoCD diff and sends notification to slack"
+inputs:
+  registry:
+    description: "Registry's host"
+    required: false
+    default: "ghcr.io"
+  user:
+    description: "Registry's user"
+    required: false
+    default: ${{ github.actor }}
+  password:
+    description: "Registry user's password or token"
+    required: true
+  image:
+    description: "Docker image name with tag"
+    required: true
+  dockle:
+    description: "Enable dockle linting steps (Default: true)"
+    required: false
+    default: true
+  argocd:
+    description: "Enable argocd diff on PRs (Default: true)"
+    required: false
+    default: true
+  argocd-auth:
+    description: "ArgoCD Auth token for production"
+    required: false
+  argocd-auth-staging:
+    description: "ArgoCD Auth token for staging"
+    required: false
+  argocd-auth-testing:
+    description: "ArgoCD Auth token for testing"
+    required: false
+  argocd-server-testing:
+    description: "ArgoCD server URI for testing"
+    required: false
+  argocd-server-staging:
+    description: "ArgoCD server URI for staging"
+    required: false
+  argocd-server:
+    description: "ArgoCD server URI for production"
+    required: false
+  slack-webhook:
+    description: "Slack Webhook for notifications"
+    required: true
+  deploy-fail-notification:
+    description: "Send notification to Slack for deploy failure"
+    required: false
+    default: false
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3
+    
+    # ==== DOCKLE LINTING ====
+    - name: Set up Docker Buildx
+      if: ${{ inputs.dockle }}
+      uses: docker/setup-buildx-action@v1
+
+    - name: Log in to GitHub Registry
+      if: ${{ inputs.dockle }}
+      uses: docker/login-action@v1.10.0
+      with:
+        registry: ${{ inputs.registry }}
+        username: ${{ inputs.user }}
+        password: ${{ inputs.password }}
+
+    - name: Dockle Security Check
+      if: ${{ inputs.dockle }}
+      uses: erzz/dockle-action@v1
+      with:
+        image: ${{ inputs.image }}
+        failure-threshold: fatal
+        exit-code: 1 # Stops pipeline if fatal issues found
+    
+    # ==== ARGOCD DIFF ====
+    - uses: imranismail/setup-kustomize@v1
+      if: ${{ inputs.argocd }}
+
+    - uses: quizlet/argocd-diff-action@master
+      name: ArgoCD Diff testing
```

Comment:
This can just be 1 argod task because in the future we would run 1 argocd diff per branch. 

Development->argocd testing
staging->argocd staging
prod-> we will see about that later :p

Reply:
Updated